libs/cardano-ledger-core/src/Cardano/Ledger/BaseTypes.hs

```diff
@@ -226,7 +226,13 @@ instance EncCBORGroup ProtVer where
   listLenBound _ = 2
 
 instance DecCBORGroup ProtVer where
-  decCBORGroup = ProtVer <$> decCBOR <*> decCBOR
+  decCBORGroup =
+    ProtVer
+      <$> decCBOR
+      <*> ifDecoderVersionAtLeast
+        (natVersion @12)
+        (fromIntegral @Word32 @Natural <$> decCBOR @Word32)
```

Comment:
Shouldn't we change CDDL.hs for dijkstra to reflect this? Or it's part of @aniketd's CDDL work?

Reply:
Yes, that's a good point. We should change that. However, because Dijkstra CDDL is still in flux, I think it will be ok to postpone until a bit later: #5414